numpy/core/src/multiarray/temp_elide.c

```diff
@@ -359,7 +359,7 @@ can_elide_temp_unary(PyArrayObject * m1)
 {
     int cannot;
     if (Py_REFCNT(m1) != 1 || !PyArray_CheckExact(m1) ||
-            PyArray_DESCR(m1)->type_num == NPY_VOID ||
+            PyArray_DESCR(m1)->type_num >= NPY_OBJECT ||
```

Comment:
the ee->f loop is only used for non fast powers which is not elided (though it could be, albeit with not much performance gain as normal pow is very slow)
adding a test for that is a good idea though